### PR TITLE
Send the FIS auth token in the RC fetch headers.

### PR DESCRIPTION
**Pull request by danasilver**

### CODE REVIEW

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClient.java [hunk 1]
```diff
@@ -78,6 +78,8 @@
   private static final String X_ANDROID_PACKAGE_HEADER = "X-Android-Package";
   private static final String X_ANDROID_CERT_HEADER = "X-Android-Cert";
   private static final String X_GOOGLE_GFE_CAN_RETRY = "X-Google-GFE-Can-Retry";
+  private static final String INSTALLATIONS_AUTH_TOKEN_HEADER =
+      "X-Goog-Firebase-Installations-Auth";
```
Comment (erikeldridge) [inline firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClient.java:82]: Thinking (no action required): this is comparable to usage of the same header in RC's iOS SDK: https://github.com/firebase/firebase-ios-sdk/blob/c07c89ee7582e84da1c5aafaa94b38a53bcfbdc2/FirebaseRemoteConfig/Sources/RCNFetch.m#L49

File: firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClientTest.java [hunk 2]
```diff
@@ -240,22 +242,22 @@ public void fetch_localeUsesToStringBelowLollipop() throws Exception {
...
   @Test
-  public void fetch_instanceIdTokenIsNull_doesNotThrowException() throws Exception {
+  public void fetch_installationAuthTokenIsNull_doesNotThrowException() throws Exception {
     setServerResponseTo(noChangeResponseBody, SECOND_ETAG);
 
-    FetchResponse fetchResponse = fetchWithoutIidToken();
+    FetchResponse fetchResponse = fetchWithoutInstallationAuthToken();
 
-    assertWithMessage("Fetch() failed with null instance id token!")
+    assertWithMessage("Fetch() failed with null installation auth token!")
```
Comment (erikeldridge) [inline firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClientTest.java:260]: Thinking (no action required): this is public error message, so the phrasing should match the public docs. The [FirebaseInstallations.getToken docs](https://firebase.google.com/docs/reference/android/com/google/firebase/installations/FirebaseInstallations#getToken(boolean)) describe "authentication token for the Firebase installation", which matches the phrasing here.

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClient.java (hunk 1 above)
Comment (erikeldridge) [reply firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClient.java:82]: ... also FIAM and FIS SDKs:
https://github.com/firebase/firebase-ios-sdk/search?q=x-goog-firebase-installations-auth&unscoped_q=x-goog-firebase-installations-auth

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/RemoteConfigConstants.java [hunk 3]
```diff
@@ -31,6 +31,10 @@
   /**
    * Keys of fields in the Fetch request body that the client sends to the Firebase Remote Config
    * server.
+   *
+   * <p>{@code INSTANCE_ID} and {@code INSTANCE_ID_TOKEN} are legacy names for the fields that used
```
Comment (ankitaj224) [inline firebase-config/src/main/java/com/google/firebase/remoteconfig/RemoteConfigConstants.java:35]: We might need some clarity here. The javadoc gives the impression that INSTANCE_ID_TOKEN can be fetched from FIS SDK, which is incorrect. We can explicitly call out INSTANCE_ID_TOKEN is replaced by  installation auth token fetched from FIS SDK.

Comment (danasilver) [reply firebase-config/src/main/java/com/google/firebase/remoteconfig/RemoteConfigConstants.java:35]: Good call. Updated.

File: firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClientTest.java (hunk 2 above)
Comment (andirayo) [reply firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHttpClientTest.java:260]: Great callout, thank you!
We had a meeting recently with DevRel and Tech Writers and decided to name the authentication tokens for Firebase installations: "installation auth tokens".
So, the action item is probably on us in the FIS team.

**Pull request merged**